index.js

```diff
@@ -30,32 +128,49 @@ function tokenIntrospect(opts = {}) {
     }
   }
 
-  return function introspect(token, tokenTypeHint) {
-    const data = { token };
-    if (tokenTypeHint) {
-      data.token_type_hint = tokenTypeHint;
+  const remoteIntrospectionInfo = Object.assign({}, options, { proxy });
+  let fetchJwks = null;
+  if (options.jwks_uri) {
+    const client = new JwksClient({
+      cache: true,
+      rateLimit: true,
+      jwksUri: options.jwks_uri,
+    });
+    fetchJwks = promisify(client.getKeys).bind(client);
+  }
+
+  return async function introspect(token, tokenTypeHint) {
+    async function tryLocalIntrospect(keys) {
```

Comment:
Split this out to local introspection file/module

Reply:
I feel this small wrapper belongs here since it's wrapping usage of the local introspection module while not actually being part of the implementation of local introspection.

Keeping it here also allows the local introspection module to only export one thing (for easier testing). But I could move it to top-level instead of nesting it inside `introspect`. What do you think?